### ISSUE TITLE
'observable' must have identical modifiers

### ISSUE THREAD

**Issue by adappter** [ReactiveX/rxjs#3892]
## Bug Report

**Current Behavior**
I get this error while npm run start:

```
    "rxjs": "6.2.1",
    "@types/node": "10.5.1"
```

```
ERROR in ../node_modules/rxjs/internal/symbol/observable.d.ts(4,9): error TS2687: All declarations of 'observable' must have identical modifiers.

node_modules/@types/node/index.d.ts(169,14): error TS2687: All declarations of 'observable' must have identical modifiers.
```

I looked up both files and they seems to be correct:
observable.d.ts
```
declare global  {
    interface SymbolConstructor {
        readonly observable: symbol;
    }
}
```

@types/node/index.d.ts
```
interface SymbolConstructor {
    readonly observable: symbol;
    readonly iterator: symbol;
    readonly asyncIterator: symbol;
}
declare var Symbol: SymbolConstructor;
```

**Expected behavior**
npm run start should work without errors

**Environment**
- Runtime: Angular 6
- RxJS version: 6.2.1

**Additional context/Screenshots**
this is the output of npm list rxjs:
```
@angular-devkit/build-angular@0.6.8
│ ├─┬ @angular-devkit/architect@0.6.8
│ │ └── rxjs@6.2.1  deduped
│ └── rxjs@6.2.1  deduped
├─┬ @angular-devkit/core@0.6.8
│ └── rxjs@6.2.1  deduped
├─┬ @angular/cli@6.0.8
│ ├─┬ @angular-devkit/schematics@0.6.8
│ │ └── rxjs@6.2.1  deduped
│ ├─┬ @schematics/update@0.6.8
│ │ └── rxjs@6.2.1  deduped
│ └── rxjs@6.2.1  deduped
└── rxjs@6.2.1
```

**Comment by cartant** [comment]
We need to establish why you are seeing an error that should have been resolved with the fix for #3769.

As you've noted, the declarations appear to be the same, so something else is going on. To take this further, a repro is necessary. Can you detail what needs to be installed to reproduce the problem? Without a repro, this isn't likely to be addressable.

I've created a new project using the Angular CLI and it starts just fine. I'm suspicious that something has been tweaked with the module resolution configuration and that your project is not using the modules that you think it is.

**Comment by adappter** [comment]
Hey cartant, 
because it's a enterprise project I can't give the repo. I just created also a new project with the same package.json as the enterprise project and there is no error.

Do you know how I can check which kind of modules my project uses? What dou you mean with "that something has been tweaked with the module resolution configuration"? 

I deleted the node modules dir and package.lock and cleaned the cache of npm.

**Comment by cartant** [comment]
>because it's a enterprise project I can't give the repo

No worries, its not the *repo* I would want in any case; I'd want a *repro* - i.e. a (minimal) reproduction of the problem. What you are doing is the approach I would take: attempt to create an minimal repro using the Angular CLI, etc.

I would look at the `tsconfig.json`, as TypeScript can be configured to resolve modules in the a number of ways. Look at the module and path-related options in the [compiler options documentation](https://www.typescriptlang.org/docs/handbook/compiler-options.html).

**Comment by adappter** [comment]
I just created a new project but I can't reproduce the problem. I copy-paste the package.json of the enterprise project.
this is what my tsconfig.json looks like
```
{
  "compileOnSave": false,
  "compilerOptions": {
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es5",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2017",
      "dom"
    ]
  },
  "angularCompilerOptions": {
    "fullTemplateTypeCheck": true
  }
}

```

**Comment by cartant** [comment]
If that's the `tsconfig.json` from the project that exhibits the problem, I cannot see anything obviously wrong with it.

At the moment, I don't have further suggestions. I guess there is something that's different between your two projects. As you're aware, you need to establish just what that is.

On the bright side, at least the new Angular CLI project works in your environment - which gives you a baseline, working project to compare against the project that exhibits the problem. If the new Angular CLI project effected the same error in your environment, but not in mine, that would be even worse.

**Comment by adappter** [comment]
Ok I will have a look. Maybe I will find something and open a new issue with the reproducible repro

**Comment by cartant** [comment]
I guess you could search in your non-working project to see if anything else augments `SymbolConstructor`. I would think that's unlikely, but you might want to check. Search and `.d.ts` or `.ts` files in that project's `node_modules`, etc.

**Comment by adappter** [comment]
Hey there,
I changed the versions of my project like this:
```
    "rxjs": "6.2.0",
    "@types/node": "10.1.2",
```
The error doesn't appear. 

If I use this versions:
```
    "rxjs": "6.2.1",
    "@types/node": "10.1.2",
```
This error appears: `node_modules/rxjs/internal/symbol/observable.d.ts(4,9): error TS2687: All declarations of 'observable' must have identical modifiers.`


If I use this versions:
```
    "rxjs": "6.2.0",
    "@types/node": "10.5.2",
```
This appears: 
```
ERROR in node_modules/@types/node/index.d.ts(169,14): error TS2687: All declarations of 'observable' must have identical modifiers.
node_modules/rxjs/internal/symbol/observable.d.ts(4,9): error TS2687: All declarations of 'observable' must have identical modifiers.
```

I checked the `node_modules` and there are only angular libs which uses `SymbolConstructor`

**Comment by cartant** [comment]
That's interesting. Your stating that this version combination fails:

`"rxjs": "6.2.1"` and `"@types/node": "10.1.4"`

Confirms that the problem is elsewhere and is the fault of another augmentation of `SymbolConstructor`.

For me, in a minimal, non-Angular CLI project (just Node and RxJS), these combinations are all fine:

`"rxjs": "6.2.1"` and `"@types/node": "10.5.2"`

`"rxjs": "6.2.1"` and `"@types/node": "10.1.4"`

`"rxjs": "6.2.0"` and `"@types/node": "10.1.4"`

Only this combination fails:

`"rxjs": "6.2.0"` and `"@types/node": "10.5.2"`

If you look at the types for `"@types/node": "10.1.4"`, you'll see that the declaration for `SymbolConstructor` does not include `observable`:

```ts
interface SymbolConstructor {
    readonly iterator: symbol;
    readonly asyncIterator: symbol;
}
declare var Symbol: SymbolConstructor;
```

So there is something, somewhere, in your project that's augmenting `SymbolConstructor` with an incorrect - i.e. non-`readonly` - declaration for `observable`.

**Comment by adappter** [comment]
I think I found the problem but I don't know if this was the solution.
What I did: 
- I installed the working version for `rxjs 6.2.0` and `@types/node 10.1.4`
- Push the working project and jenkins started to build
- It failed at the test pipeline with the error message, that I'm using wrong imports e.g. `rxjs/add/observable/of`
- Fixing all tests with correct imports, e.g. `import {  of } from "rxjs";`
- Pushed again and the build worked

It seems that the path of the imports causes this error.
